packages/commerce-sdk-react/src/test-utils.tsx

```diff
@@ -6,8 +6,7 @@
  */
 
 import React from 'react'
-import {render, RenderOptions} from '@testing-library/react'
-import {renderHook, WaitForValueToChange} from '@testing-library/react-hooks/dom'
```

Comment:
This library is not compatible with React 18, the `renderHook` has been moved to testing-library/react, but the Result api has changed, we need to adjust our test to not use `waitForValueToChange`

> As part of the changes for React 18, it has been decided that the renderHook API provided by this library will instead be included as official additions to both react-testing-library ([PR](https://github.com/testing-library/react-testing-library/pull/991)) and react-native-testing-library ([PR](https://github.com/callstack/react-native-testing-library/pull/923)) with the intention being to provide a more cohesive and consistent implementation for our users.
> 
> Please be patient as we finalise these changes in the respective testing libraries. In the mean time you can install @testing-library/react@^13.1